### PR TITLE
Convert tagging information to index/log

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Store detailed tagging information in a log at command buffer scope, and
only store the index in the resource state.

@jeremyg-lunarg -- stripped out from the secondary command buffers change per your request.

### CODE REVIEW

File: layers/synchronization_validation.cpp [hunk 1]
```diff
@@ -1143,7 +1143,7 @@ class HazardDetector {
 
   public:
     HazardResult Detect(const ResourceAccessRangeMap::const_iterator &pos) const { return pos->second.DetectHazard(usage_index_); }
-    HazardResult DetectAsync(const ResourceAccessRangeMap::const_iterator &pos, const ResourceUsageTag &start_tag) const {
+    HazardResult DetectAsync(const ResourceAccessRangeMap::const_iterator &pos, ResourceUsageTag start_tag) const {
```
Comment (ncesario-lunarg) [inline layers/synchronization_validation.cpp:1146]: Just to clarify, these (and similar instances in this PR) are _still_ `const`, just not references anymore?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

File: layers/synchronization_validation.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/synchronization_validation.cpp:1146]: I took this change as expected since the tag type changed from a struct to an integer type.

Comment (jzulauf-lunarg) [reply layers/synchronization_validation.cpp:1146]: @ncesario-lunarg -- and when the definition *is* the declaration (which in this case just feels like lazy typing on my, and we're not consistent about "how big is too big for inline definition) you can't const it for the body.

**Pull request merged**